src/java/org/jivesoftware/openfire/SessionManager.java

```diff
@@ -641,10 +641,12 @@ public void broadcastPresenceToOtherResources(JID originatingResource, Presence
         JID searchJID = new JID(originatingResource.getNode(), originatingResource.getDomain(), null);
         List<JID> addresses = routingTable.getRoutes(searchJID, null);
         for (JID address : addresses) {
-            // Send the presence of the session whose presence has changed to
-            // this other user's session
-            presence.setTo(address);
-            routingTable.routePacket(address, presence, false);
+        	if (originatingResource != address) {

```

Comment:
I guess @wmz7year meant that `!equals()` should be used instead of a comparison of the object references.